NOTE-act-rules-common-aspects.bs

```diff
@@ -69,3 +69,18 @@ Language, either written or spoken, contained in nodes of the DOM [[DOM]] or acc
 Rules can only operate on the Language aspect if the language of a page can be determined. The means by which the language is assessed, whether by a person or a machine, is not of importance as long as the assessment meets the criteria defined in [[wcag2-tech-req#humantestable]] [[WCAG]].
 
 Some rules can only operate on a Language aspect if the language is sufficiently understood, while other rules only require identifying the language. For example, a rule checking that an accessible name is descriptive can only function if the language is understood. A rule checking the correctness of a `lang` attribute requires knowing what language is used, but not the meaning of the words.
+
+Source Code {#input-aspects-code}
+----------------------------------
+
+Source code, the programming language used to build a web page, may be of interest to ACT Rules that intend to test things like the presence of certain HTML elements or attributes. For example, an ACT Rule might test that elements do not contain duplicate attributes.
```

Comment:
I think this has some important edge cases that I'd like to be more explicit about. 

```suggestion
The byte data of a file from which a web browser or other user agent creates a page. For example, a browser may build up a web page from an HTML file, CSS file, and JavaScript file. The text of each of these three files is its source code. An ACT Rule could for example test for potential parser errors.

Source code is distinct from an HTTP response, which includes HTTP headers. It is the byte content of the file before parsing, which often results in an object model or syntax tree, or serialised versions of those. Notably in HTML, the outerHTML property of the root node can vary significantly from its source code. For the purpose of ACT rules, source files used in preprocessing such as PHP for creating HTML, or SASS for creating CSS are not considered, as those are not processed by the web browser.
```

Reply:
`HTML file, CSS file, and JavaScript file` -> `HTML file, CSS file, JavaScript File, SVG Image or PDF`